fictadvisor-api/src/v2/api/controllers/UserController.ts

```diff
@@ -354,113 +136,42 @@ export class UserController {
     return { selectives: await this.userService.getSelectivesBySemesters(userId) };
   }
 
-  @ApiBearerAuth()
-  @ApiOkResponse()
-  @ApiBadRequestResponse({
-    description: `\n
-    InvalidBodyException: 
-      Role id cannot be empty
-    
-    InvalidEntityIdException:
-      User with such id is not found`,
-  })
-  @ApiUnauthorizedResponse({
-    description: `\n
-    UnauthorizedException:
-      Unauthorized`,
-  })
-  @ApiForbiddenResponse({
-    description: `\n
-    NoPermissionException:
-      You do not have permission to perform this action`,
-  })
-  @ApiParam({
-    name: 'userId',
-    required: true,
-    description: 'Id of a user',
-  })
   @ApiEndpoint({
     summary: 'Give role to student',
+    documentation: UserDocumentation.GIVE_ROLE,
     guards: TelegramGuard,
   })
-  @Post('/:userId/roles')
-  giveRole (
+  @Patch('/:userId/roles')
+  async giveRole (
     @Param('userId', UserByIdPipe) userId: string,
     @Body() { roleId }: GiveRoleDTO,
   ) {
+    await this.roleService.checkRoleExists(roleId);
     return this.userService.giveRole(userId, roleId);
   }
```

Comment:
For consistency it would be better to write it this way
```suggestion
  @Patch('/:userId/roles')
  async giveRole (
    @Param('userId', UserByIdPipe) userId: string,
    @Body('roleId', RoleByIdPipe) roleId: string,
    @Body() body: GiveRoleDTO,
  ) {
    return this.userService.giveRole(userId, roleId);
  }
```